File: SemanticNetwork/src/main/java/de/kimanufaktur/nsm/graph/entities/marker/DoubleMarker.java
/*
 * Copyright (C) Johannes Fähndrich - All Rights Reserved.
 * Unauthorized copying of this file, via any medium is strictly
 * prohibited Proprietary and confidential.
 * Written by Johannes Fähndrich <faehndrich@gmail.com.com>,  2011
 */

package de.kimanufaktur.nsm.graph.entities.marker;

import de.kimanufaktur.markerpassing.Marker;

/**
 * Created by faehndrich on 19.05.15.
 */
public class DoubleMarker implements Marker {
    double activation = Double.NaN;

    public double getActivation() {
        return activation;
    }

    public void setActivation(double activation) {
        this.activation = activation;
    }
}

File: semantic_decomposition/entities/markers/double_marker.py
from ...marker_passing.marker import Marker


class DoubleMarker(Marker):
    """Marker carrying a single floating-point activation value."""

    def __init__(self, activation: float = 0.0) -> None:
        self.activation: float = activation

    def __repr__(self) -> str:
        return f"DoubleMarker({self.activation})"
